### PR TITLE
[No QA] Fix/52558 fix card name and images

### PR DESCRIPTION
**Pull request by koko57**
### Fixed Issues

$ https://github.com/Expensify/App/issues/52558
PROPOSAL:

### Tests

IMAGES
Check if the card icon sizing is appropriate on: 
- Card List page (feed selector), 
- Feed Selector page, 
- Members -> Member Details -> New Card page (accessed after clicking New Card)
- Wallet main page (cards section)

CARD NAME
- verify that the "[FirstName]'s card" is displayed as a fallback for the cards that had no name displayed after issuing
- verify that the newly created Expensify Cards will have the "card" lowercased

- [x] Verify that no errors appear in the JS console

### QA Steps

// TODO: These must be filled out, or the issue title must include "[No QA]."

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 39 14
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 39 52

Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 39 06

Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 41 25
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 41 22
</details>

<details>
<summary>iOS: mWeb Safari</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-11-20 at 20 23 57
Simulator Screenshot - iPhone 15 Pro - 2024-11-20 at 20 23 51
Simulator Screenshot - iPhone 15 Pro - 2024-11-20 at 20 24 14

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-11-20 at 19 32 09
Screenshot 2024-11-20 at 19 29 16

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-11-20 at 20 32 45
Screenshot 2024-11-20 at 20 32 24
Screenshot 2024-11-20 at 20 32 12

</details>

### CODE REVIEW

**Comment by dannymcclain** [comment]
>Should the card have rounded edges?
>I would think yes - not sure if that's something we should do on the svg itself or if that comes from a style we apply?

@shawnborton 🤔 Hmm, interesting. It looks like the SVGs _do_ have rounded corners?
CleanShot 2024-11-20 at 13 27 17@2x

**Comment by koko57** [comment]
@dannymcclain we no longer use the icons from the company cards folder - we use the ones saved in companycards/large

still, it's strange as the icons from the 'large' folder also have the corners rounded

cc @mountiny

**Comment by koko57** [comment]
@dannymcclain @shawnborton what is the appropriate border radius for the small card icons - in the lists?

**Comment by koko57** [comment]
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 35 12

after the lates changes

**Comment by koko57** [comment]
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 41 25
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 41 22
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 39 06
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 39 14
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 09 39 52

**Comment by shawnborton** [comment]
Nice, that's feeling better! Looks like we're using `4` for the border radius of the small cards in Figma.

**Comment by koko57** [comment]
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 19 15 46
Screenshot 2024-11-21 at 19 15 39

@shawnborton fixed

**Review by allgandalf — COMMENTED** [review]
okay changes LGTM, testing now

**Comment by koko57** [comment]
@mountiny and what about this one? https://github.com/Expensify/App/pull/52735#issuecomment-2491116917

**Comment by koko57** [comment]
Screenshot 2024-11-22 at 10 12 01

I fixed the problem with the cards in Search 

cc @joekaufmanexpensify

**Comment by koko57** [comment]
@allgandalf addressed

File: src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx
Comment (mountiny) [reply src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx:61]: I think we should truncate for such a long names, not much else to do here

**Review by allgandalf — APPROVED** [review]
LGTM and tests well! Great work here @koko57 🙇

File: src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx
Comment (mountiny) [reply src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx:61]: Just waiting for this discussion

**Review by mountiny — DISMISSED** [review]
LGTM except waiting for confirmation to truncate it. @koko57 I think you can try to add this in now

**Comment by koko57** [comment]
@mountiny conflicts resolved. Text truncated
Screenshot 2024-11-22 at 20 39 01

**Comment by koko57** [comment]
cc @allgandalf

File: src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx
Comment (shawnborton) [reply src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx:60]: Good catch!

**Comment by dannymcclain** [comment]
>For the truncation part... is it better to just show the last 4 digits on mobile? Otherwise you basically see the same exact numbers there and it isn't super helpful.

Agree. I'm not sure we can truncate the last 4 here. @shawnborton do you think it'd be better to:
- _never_ truncate the number and truncate more of the name
- do both like we're doing here but truncate the _beginning_ of the card number instead of the end
- just show last 4 on mobile like you suggested?

**Comment by shawnborton** [comment]
> do both like we're doing here but truncate the beginning of the card number instead of the end

I could get down with something like this

**Comment by dannymcclain** [comment]
🤔 Hmm, if we're considering truncating the beginning of the card number... Is there any reason not to just go ahead and only show the last 4? If you can't see the beginning, are the middle card numbers particularly helpful at all?

**Comment by koko57** [comment]
Screenshot 2024-11-25 at 10 08 58

I also changed the header to "Last 4" instead of "Card Number" when only 4 last digits are displayed for mobile/narrow screen.

**Comment by koko57** [comment]
@allgandalf fixed

**Review by mountiny — DISMISSED** [review]
Discussing in Slack the change to last 4 pan numbers before merging

**Comment by mountiny** [comment]
I think that is good!

**Comment by koko57** [comment]
If any other changes will be necessary, I suggest to do it as a follow-up PR as it'll be more changes in the code.

**Review by mountiny — APPROVED** [review]
discussed and for now this is good. We might follow up with changing how we show the card numbers across the app later.

**Comment by joekaufmanexpensify** [comment]
> IMAGES
> Check if the card icon sizing is appropriate on:
> Card List page (feed selector),
> Feed Selector page,
> Members -> Member Details -> New Card page (accessed after clicking New Card)
> Wallet main page (cards section)

Art looks good everywhere! 

image
image
image
image

**Comment by joekaufmanexpensify** [comment]
> CARD NAME
> verify that the "[FirstName]'s card" is displayed as a fallback for the cards that had no name displayed after issuing

Confirmed. 
image
image

**Comment by joekaufmanexpensify** [comment]
> verify that the newly created Expensify Cards will have the "card" lowercased

Not sure where I am supposed to check this. Asking!

**Comment by joekaufmanexpensify** [comment]
[Asked](https://expensify.slack.com/archives/C06ML6X0W9L/p1732651175678639)

**Comment by koko57** [comment]
@joekaufmanexpensify for the physical Expensify Card if you don't change the default name during the issuing process

**Comment by joekaufmanexpensify** [comment]
Got it. Tested and confirmed default name is `%Firstname's% card` and card is lowercased.

**Pull request merged**
